src/library/coverartutils.cpp

```diff
@@ -29,13 +29,8 @@ QString CoverArtUtils::supportedCoverArtExtensionsRegex() {
 QImage CoverArtUtils::extractEmbeddedCover(
         const QString& trackLocation,
         SecurityTokenPointer pToken) {
-    SoundSourceProxy proxy(trackLocation, pToken);
-    QImage coverArt;
-    if (OK == proxy.parseTrackMetadataAndCoverArt(nullptr, &coverArt)) {
-        return coverArt;
-    } else {
-        return QImage();
-    }
+    TrackPointer pTrack(new TrackInfoObject(trackLocation, pToken));
```

Comment:
I think we should use a Track from the track cache to ensure we have only one TrackObject per file. 
This is probably a topic of a future PR. A comment would be nice. 

trackLocation() implicit converts a Qstring to a QFileInfo, It looks like the caller already have an QFileInfo so we should pass it through.